### ISSUE TITLE
[rllib] Jenkins test failure "cannot import name spaces".

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#1486]
the following test failure seems to have started recently (and is deterministic?) in 

```
docker run --rm --shm-size=10G --memory=10G 5856c64c69ea8839954e3e0384a95122d118d54439be82b30d22dacfae013a19 python /ray/python/ray/rllib/train.py --env PongDeterministic-v0 --run A3C --stop '{"training_iteration": 2}' --config '{"num_workers": 16}'
```

```
WARNING: Your kernel does not support swap limit capabilities or the cgroup is not mounted. Memory limited without swap.
Allowing the Plasma store to use up to 8.58993GB of memory.
Starting object store with directory /dev/shm and huge page support disabled
Traceback (most recent call last):
Waiting for redis server at 127.0.0.1:33097 to respond...
Waiting for redis server at 127.0.0.1:22106 to respond...
Warning: Reducing object store memory because /dev/shm has only 10737418240 bytes available. You may be able to free up space by deleting files in /dev/shm. If you are inside a Docker container, you may need to pass an argument with the flag '--shm-size' to 'docker run'.
Starting local scheduler with the following resources: {'GPU': 0, 'CPU': 48}.

======================================================================
View the web UI at http://localhost:8888/notebooks/ray_ui93627.ipynb?token=6ad9d63780ce4768845e0c6b19720dd26644ff6dd677f849
======================================================================

Warning: could not import PPO: cannot import name spaces
Warning: could not import ES: cannot import name error
Warning: could not import DQN: cannot import name error
Warning: could not import A3C: cannot import name error
Warning: could not import BC: cannot import name error
  File "/ray/python/ray/rllib/train.py", line 83, in <module>
    run_experiments(experiments, scheduler=_make_scheduler(args))
  File "/opt/conda/lib/python2.7/site-packages/ray-0.3.0-py2.7-linux-x86_64.egg/ray/tune/tune.py", line 46, in run_experiments
    for trial in generate_trials(spec, name):
  File "/opt/conda/lib/python2.7/site-packages/ray-0.3.0-py2.7-linux-x86_64.egg/ray/tune/variant_generator.py", line 65, in generate_trials
    upload_dir=args.upload_dir)
  File "/opt/conda/lib/python2.7/site-packages/ray-0.3.0-py2.7-linux-x86_64.egg/ray/tune/trial.py", line 90, in __init__
    raise TuneError("Unknown trainable: " + trainable_name)
ray.tune.error.TuneError: Unknown trainable: A3C
Disconnecting client on fd 11
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 27 which hadn't started yet.
[WARN] (/ray/src/global_scheduler/global_scheduler.cc:400) Missed too many heartbeats from local scheduler, marking as dead.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 28 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 29 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 30 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 31 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 32 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 33 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 34 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 35 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 36 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 37 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 38 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 39 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 40 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 41 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 42 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 43 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 44 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 45 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 46 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 47 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 48 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 49 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 50 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 51 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 52 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 53 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 54 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 55 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 56 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 57 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 58 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 59 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 60 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 61 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 62 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 63 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 64 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 65 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 66 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 67 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 68 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 69 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 70 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 71 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 72 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 73 which hadn't started yet.
[INFO] (/ray/src/local_scheduler/local_scheduler.cc:171) Killed worker pid 74 which hadn't started yet.
Disconnecting client on fd 9
Disconnecting client on fd 7
Build step 'Execute shell' marked build as failure
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/3470/
Test FAILed.
Finished: FAILURE
```

For an example, see https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/3475/.

cc @ericl @richardliaw

**Comment by robertnishihara** [comment]
The issue only seems to occur with Python 2.7.

**Comment by pcmoritz** [comment]
@richardliaw You can try this fix: https://github.com/openai/gym/pull/851

**Comment by richardliaw** [comment]
Temporary fix in #1490.
